src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -248,6 +242,12 @@ HashSet<ITypeSymbol> typeStack
 				return MutabilityInspectionResult.NotMutable();
 			}
 
+			var scope = type.GetImmutabilityScope();
+			if( !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute ) && scope != ImmutabilityScope.None ) {

```

Comment:
Safe to assume `!= Immutability.None` will always be a valid scope here? Or better to be explicit about which scopes you want to test? I'm fine with either way.

Reply:
Good point. I'm very confident that enum won't ever gain/lose members. I'm going to go with this for now. I might change how that stuff is structured 9the enum/helpers) after the flags are gone but still undecided.